Source/Dafny/Verifier/Translator.cs

```diff
@@ -11142,6 +11142,12 @@ public enum K { Free, Checked, Both }
         }
 
         return true;
+
+      } else if (expr is MatchExpr) {
+        var e = (MatchExpr)expr;
```

Comment:
Please put the name `e` in the if condition. Also, I suggest you rename it to `matchExpr` since the difference between `e` and `expr` is unclear from the names.